RELEASING.md

```diff
@@ -259,36 +286,35 @@ as described in the section above.
    make release-publish
    ```
 
-1. Merge the PR. This will cause candidate.docs.projectcalico.org to be updated (after a few minutes). Validate that everything looks correct before proceeding to the next step.
+1. Merge the PR. This will cause docs.projectcalico.org to be updated (after a few minutes). Validate that everything looks correct.
 
-1. Checkout the master branch
+1. Remove branch deploy for the candidate branch in docs.projectcalico.org site settings.
```

Comment:
I thought we left the branch deploy in place for the branch as it transitions from candidate to latest?